merge.go

```diff
@@ -266,9 +266,9 @@ var (
 	//_ RowWriterTo         = (*mergedRowGroupRowReader)(nil)
 )
 
-type sortFunc func(a, b []Value) int
+type SortFunc func(a, b []Value) int
 
-func sortFuncOf(t Type, maxDefinitionLevel, maxRepetitionLevel int8, descending, nullsFirst bool) (sort sortFunc) {
+func SortFuncOf(t Type, maxDefinitionLevel, maxRepetitionLevel int8, descending, nullsFirst bool) (sort SortFunc) {
```

Comment:
I'm a bit concerned about exposing this API as-is, it doesn't really follow good practices of being hard to misuse: for example, there are two boolean parameters which say nothing about what they are at the call site (e.g. `parquet.SortFuncOf(typ, d, r, false, false)`), the consecutive arguments of same types also make it easy to mistakenly swap the parameters and introduce bugs that compilers or linters cannot detect.

Let me circle back on this with a proposal later today.

Reply:
Yeah, that makes perfect sense. Let me know if you come up with an interface you like!

Reply:
> the consecutive arguments of same types also make it easy to mistakenly swap the parameters and introduce bugs that compilers or linters cannot detect.

As a matter of fact, there was actually a bug in the implementation where the repetition and definition levels were inverted in the call to `sortFunc` 🤦  I should have known to apply my own advises!

This should be fixed in https://github.com/segmentio/parquet-go/pull/93